core/src/main/java/org/infinispan/container/offheap/BoundedOffHeapDataContainer.java

```diff
@@ -257,12 +257,16 @@ private void ensureSize() {
                log.tracef("Removing entry: %d due to eviction due to size %d being larger than maximum of %d",
                      addressToRemove, currentSize, maxSize);
             }
+            InternalCacheEntry<WrappedBytes, WrappedBytes> ice = null;
             try {
-               InternalCacheEntry<WrappedBytes, WrappedBytes> ice = offHeapEntryFactory.fromMemory(addressToRemove);
+               ice = offHeapEntryFactory.fromMemory(addressToRemove);
                performRemove(addressToRemove, offHeapEntryFactory.getKey(addressToRemove));
                evictionManager.onEntryEviction(Collections.singletonMap(ice.getKey(), ice));
             } finally {
                entryWriteLock.unlock();
+               if (ice != null) {
+                  passivator.passivate(ice);
```

Comment:
Passivation potentially invokes listeners and throws an exception, so I would put it after the finally block.